receiver.go

```diff
@@ -45,16 +45,13 @@ func (r *Receiver) HandleMessage(ctx context.Context, handle func(*Message) erro
 	callHandler := func(msg *Message) error {
 		debug(3, "Receive() blocking %d", msg.deliveryID)
 		msg.receiver = r
-		if msg.doneSignal == nil {
-			msg.doneSignal = make(chan struct{})
-		}
-		go trackCompletion(msg)
-		// accept spontaneously when receiver is in ModeFirst
+		// we only need to track message disposition for mode second
 		// spec : http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#type-receiver-settle-mode
-		if r.link.receiverSettleMode.value() == ModeFirst {
-			if err := msg.Accept(ctx); err != nil {
```

Comment:
We still need to Accept the msg in modeFirst. is that happening somewhere else?

Reply:
We only needed to accept it to close the channel, but since we no longer create the channel for mode first it's no longer required.

Reply:
no, in mode first, the client must spontaneously accept the message on reception, that's why I was calling Accept here.
we need to settle it on behalf of the consumer

Reply:
PR was linking to the spec : 
https://github.com/Azure/go-amqp/pull/26

Reply:
Hmm that's what I thought too, however in my testing I observed that `m.shouldSendDisposition()` returned `false` so no disposition was sent.  I will investigate further.

Reply:
that would mean that the message is sent as settled.
I thought that's what I verified wasn't true. ha
we can look at the frames being sent by SB in receivemode 1 and see if the message is indeed pre-settled.

Reply:
ok, verified, it comes as pre-settled: 

12:57:54.818175 RX(Session): Transfer{Handle: 0, DeliveryID: 0, DeliveryTag: "", MessageFormat: 0, **Settled: true**, More: false, ReceiverSettleMode: <nil>, State: <nil>, Resume: false, Aborted: false, Batchable: true, Payload [size]: 270}